ds18S20 1-Wire sensor support

deisi opened the pull request:
- add ds18S20 1-Wire sensor support
- gitignore emacs backup files

Commit:
- add ds18S20 1-Wire sensor support
- gitignore emacs backup files

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+    try:
+        ## only one name given
+        if type(config['names']) == str:
+            names = [names]
```
I think `[names]` should be `config[names]` here ?

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the one wire Sensors"""
+    # TODO check if kernel modules are loaded
```
If the kernel module is not loaded, will the files still show up in `/sys/bus/w1/devices` folder?

The style tests do not allow TODO to be added to the codebase. If it is an idea and not something that still has to be done to get it working, replace it with FUTURE.

deisi replied:
As far as I know, the files do not show up, but I have not tested it. My
Idea was to give the user a hint, why it is not working, if the sensors
can not be loaded.

On 29.03.2015 17:50, Paulus Schoutsen wrote:
>> +import os
>> +import time
>> +import logging
>> +
>> +
>> +BASE_DIR = '/sys/bus/w1/devices/'
>> +DEVICE_FOLDERS = glob(os.path.join(BASE_DIR, '28*'))
>> +SENSOR_IDS = [os.path.split(device_folder)[1] for device_folder in DEVICE_FOLDERS]
>> +DEVICE_FILES = [os.path.join(device_folder, 'w1_slave') for device_folder in DEVICE_FOLDERS]
>> +
>> +_LOGGER = logging.getLogger(__name__)
>> +
>> +# pylint: disable=unused-argument
>> +def setup_platform(hass, config, add_devices, discovery_info=None):
>> +    """ Sets up the one wire Sensors"""
>> +    # TODO check if kernel modules are loaded
> If the kernel module is not loaded, will the files still show up in `/sys/bus/w1/devices` folder?
>
> The style tests do not allow TODO to be added to the codebase. If it is an idea and not something that still has to be done to get it working, replace it with FUTURE.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/balloob/home-assistant/pull/78/files#r27355580

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+
+        ## map names with ids
+        elif type(config['names']) == dict:
+            for sensor_id in SENSOR_IDS:
```
You could fallback to use the id as the name if the id key is not found in the dict:
```python
names = [config['names'].get(sensor_id, sensor_id) for sensor_id in SENSOR_IDS]
```
(the second parameter to get is the fallback value if key not found)

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+    names = []
+    try:
+        ## only one name given
+        if type(config['names']) == str:
```
Preferred way to type check is `isinstance(config['names'], str)` because it takes inheritance into account.

balloob commented:
Thanks for the pull request. I had some comments and it looks like the style checker had too (see Travis CI build details).

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+            for sensor_id in SENSOR_IDS:
+                names.append(sensor_id)
+        
+    for device_file, name in zip(DEVICE_FILES, names):
```
Optional: these three lines could be replaced with a list comprehension

deisi replied:
By now I'm back with this project and willing to finally bring it to an end. I see you have already suggested a lot of changes. Now my github skills are not as they should be. What do I need to do to put in your suggested changes, or are they already in this pull request? Do I need to make new pull request? I'm a bit confuesd by this github stuff here.

Commit:
little bug

deisi committed:
gitignore ready for emacs

deisi committed:
Update .gitignore

deisi committed:
Changed handling of config file 

I tried to implement your suggesteions for the default handlig of the device names. I think this way, everything you wanted is in.

deisi committed:
Improved onewire configuration

deisi committed:
merged with current dev

deisi committed:
updated with remote

deisi committed:
respect pylint suggestions

deisi committed:
respect flake8 errors

deisi committed:
new try to add everythin from the stylecheckers

deisi commented:
Any change of getting this into the offical branch? Is there anything else I need to do? I think the checks are passed so far.

balloob commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+    @property
+    def state(self):
+        """ return temperature in unit_of_measurement"""
+        lines = self._read_temp_raw()
```
Updating the state should happen in the `update` method and it should store the state in an instance variable that `state()` will return.

balloob commented:
@deisi sorry, I didn't realize you have updated this PR. I only get messages about new comments!

The code looks good :dolphin: Just one minor comment about how to handle state.

philipbl commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+class OneWire(Entity):
+    """ A Dallas 1 Wire Sensor"""
+
+    def __init__(self, name, device_file, unit_of_measurement):
```
The only value `unit_of_measurement` gets set to is `TEMP_CELCIUS` so it looks like `unit_of_measurement` can be removed as a parameter.

philipbl commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+        if equals_pos != -1:
+            temp_string = lines[1][equals_pos+2:]
+            temp = float(temp_string) / 1000.0
+            if self._unit_of_measurement == TEMP_FAHRENHEIT:
```
If `unit_of_measurement` is always `TEMP_CELCIUS`, then there is no need for this check.

philipbl commented on homeassistant/components/sensor/onewire.py:
```diff
@@ -0,0 +1,93 @@
+    names = SENSOR_IDS
+
+    for key in config.keys():
+        if key == "names":
```
No need to loop over the config keys. Just use 

```python
if "names" in config:
```

sfam commented:
I've tested this PR with one DS18S20 in parasitic mode with a 10 meter cable. Works fine, but sometimes I get values incorrect temperatures (-2000ºC , 1500ºC, etc...) but with CRC OK

```
33 81 a5 d1 ef df 00 81 f0 : crc=f0 YES
33 81 a5 d1 ef df 00 81 f0 t=-2028812

a8 60 64 f4 f7 cf 00 c1 f1 : crc=f1 YES
a8 60 64 f4 f7 cf 00 c1 f1 t=1546500

9d c0 c8 e8 ef ff 00 62 e8 : crc=e8 YES
9d c0 c8 e8 ef ff 00 62 e8 t=-1014187
```


This is messing up the graphs:


![image](https://cloud.githubusercontent.com/assets/12089995/12328723/8ab84060-bad4-11e5-9feb-e328548466e2.png)


Maybe we should ignore values outside supported range: [ -55°C to +125°C ]

outofmyleague commented:
Hi, 

I am completely new to home assistant. Its been years since I have done anything with computers and I'm feeling overwhelmed, I am a builder, not a programmer. I am very interested in home automation and want a customisable system.  I have been reading a few books and trying some sample python code.

Home Assistant seems to be exactly what I'm looking for as I am currently building my new house. I want a system to monitor individual room temps, external temps, primary boiler temp, secondry boiler temp, control windows and doors to moderate temps, control the primary air inlet for the sealed wood burner, alarm control, cctv, link into my sonos system and have preset lighting scenes.

I can have the pi centrally located, so that the 10m cable limit shouldn't be a problem. 

Sorry for the long intro...    So I have added the onewire.py and the .gitignore bits to the relevant folders, but what do I need to add to the configuration.yaml to get this to work.

I have been trying for the past 5 hours and cant get anything more that an orange circle near the sun one. I cannot get the temperature to show.

I have tested the sensor periodically to ensure its still working and I can draw the data from it.   

Thanks reading, any help would be appreciated.

outofmyleague commented:
Been testing some automation's with a few of these sensors running in parallel. Everything seems to be running well. 

One question: 

What bit of onewire.py  will i have to alter so home assistant shows each sensor with a "user defined" name. The code is functional as it is, but looks odd having a 14 digit code instead of say "living room or boiler/cyclinder" on the desktop.

balloob commented:
From the config it looks like you can do something like this:
```yaml
sensor:
  platform: onewire
  names:
    some_id: your name
```

outofmyleague commented:
Thanks Paulus

 

Any idea how I could limit the Temperatures to 1 decimal place. Currently showing up to 3.

 

Also thanks for your help, really looking forward rigging up the full system. I want to have an idea of what cables to run through my house before we finish 1st fix and start plastering.

 

 

 

 

 

From: Paulus Schoutsen [mailto:notifications@github.com] 
Sent: 27 January 2016 04:57
To: balloob/home-assistant
Cc: outofmyleague
Subject: Re: [home-assistant] ds18S20 1-Wire sensor support (#78)

 

From the config it looks like you can do something like this:

sensor:
  platform: onewire
  names:
    some_id: your name

—
Reply to this email directly or view it on GitHub <https://github.com/balloob/home-assistant/pull/78#issuecomment-175399054> .  <https://github.com/notifications/beacon/AQEtYw8J8DFpTkzyHA-A3QUvggUXPAowks5peEWegaJpZM4D2o3z.gif>

deisi commented:
Hey

Does it work with multiple sensors? I have never tested it. For the 
digits, I could add a round method.

Regards
On 1/27/2016 8:31 AM, outofmyleague wrote:
> Thanks Paulus
>
>
>
> Any idea how I could limit the Temperatures to 1 decimal place. 
> Currently showing up to 3.
>
>
>
> Also thanks for your help, really looking forward rigging up the full 
> system. I want to have an idea of what cables to run through my house 
> before we finish 1st fix and start plastering.
>
>
>
>
>
>
>
>
>
>
>
> From: Paulus Schoutsen [mailto:notifications@github.com]
> Sent: 27 January 2016 04:57
> To: balloob/home-assistant
> Cc: outofmyleague
> Subject: Re: [home-assistant] ds18S20 1-Wire sensor support (#78)
>
>
>
> From the config it looks like you can do something like this:
>
> sensor:
> platform: onewire
> names:
> some_id: your name
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/balloob/home-assistant/pull/78#issuecomment-175399054> 
> . 
> <https://github.com/notifications/beacon/AQEtYw8J8DFpTkzyHA-A3QUvggUXPAowks5peEWegaJpZM4D2o3z.gif> 
>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/balloob/home-assistant/pull/78#issuecomment-175464038>.
>

outofmyleague commented:
Hi deisi

I have it running with 2 at the minute, I plan to have 12. Waiting on a delivery.

It would better for me to have less decimals.

Thanks
________________________________
From: deisi<mailto:notifications@github.com>
Sent: ‎27/‎01/‎2016 08:33
To: balloob/home-assistant<mailto:home-assistant@noreply.github.com>
Cc: outofmyleague<mailto:slocombematthew@hotmail.com>
Subject: Re: [home-assistant] ds18S20 1-Wire sensor support (#78)

Hey

Does it work with multiple sensors? I have never tested it. For the
digits, I could add a round method.

Regards
On 1/27/2016 8:31 AM, outofmyleague wrote:
> Thanks Paulus
>
>
>
> Any idea how I could limit the Temperatures to 1 decimal place.
> Currently showing up to 3.
>
>
>
> Also thanks for your help, really looking forward rigging up the full
> system. I want to have an idea of what cables to run through my house
> before we finish 1st fix and start plastering.
>
>
>
>
>
>
>
>
>
>
>
> From: Paulus Schoutsen [mailto:notifications@github.com]
> Sent: 27 January 2016 04:57
> To: balloob/home-assistant
> Cc: outofmyleague
> Subject: Re: [home-assistant] ds18S20 1-Wire sensor support (#78)
>
>
>
> From the config it looks like you can do something like this:
>
> sensor:
> platform: onewire
> names:
> some_id: your name
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/78#issuecomment-175399054>
> .
> <https://github.com/notifications/beacon/AQEtYw8J8DFpTkzyHA-A3QUvggUXPAowks5peEWegaJpZM4D2o3z.gif>
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/78#issuecomment-175464038>.
>



---
Reply to this email directly or view it on GitHub:
https://github.com/balloob/home-assistant/pull/78#issuecomment-175479507
